### PR TITLE
Fix some bugs in syncpack configuration

### PR DESCRIPTION
**Pull request by samueljseay**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This fixes some bugs in the syncpack configuration and also fixes some packages to specific versions that were not before.

This fixes some build breaking issues I was seeing in a recent PR of mine.

### How to test the changes in this Pull Request:

1. Ensure the build still works: `TURBO_FORCE=true pnpm run build` from project root
2. CI checks passing should confirm everything else needed to test here (lint etc)

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by samueljseay** [comment]
@psealock These are some changes I needed to get builds passing in my live branches PR, but they really need to be a separate PR so here it is.

File: .syncpackrc [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"dev": true,
-	"filter": "^(?:react|react-dom|typescript)$",
+	"filter": "^(?:react|react-dom|typescript|@typescript-eslint|@types/react).*$",
```
Comment (samueljseay) [inline .syncpackrc:3]: This filter wasn't working before, so the `@typescript-eslint/**` dependency group wasn't being version matched. the `.*` ensures that monorepo packages are matched. (e.g. `@typescript-eslint/some-package` )

I find the filter kind of a weird API, It'd be nice if you could just have syncpack care about dependency matchers that you add in the `versionGroups` property, but I discovered it doesn't work that way.

File: package.json [hunk 2]
```diff
@@ -49,7 +49,7 @@
 		"request": "^2.88.2",
 		"sass": "^1.49.9",
 		"sass-loader": "^10.2.1",
-		"syncpack": "^8.2.4",
+		"syncpack": "^8.3.9",
```
Comment (samueljseay) [inline package.json:52]: Updated syncpack to latest

**Review by psealock — APPROVED** [review]
Nice fixup @samueljseay, thanks for the changes. All looking good

File: .syncpackrc (hunk 1 above)
Comment (JamieMason) [reply .syncpackrc:3]: Can you open an issue @samueljseay please? Would be happy to take a look

Comment (samueljseay) [reply .syncpackrc:3]: @JamieMason Sure I'll try and come up with a coherent suggestion 😄

Comment (jonathansadowski) [reply .syncpackrc:3]: @samueljseay seeing this late, but I noticed this while looking into something else with syncpack. It looks like this change would also match `react.*`, etc (which makes `react-dom` redundant, for example).

What do you think about moving the .* into the group where it's needed? For example:
```suggestion
	"filter": "^(?:react|react-dom|typescript|@typescript-eslint.*|@types/react)$",
```

Comment (JamieMason) [reply .syncpackrc:3]: I've not tested this but the following config should hopefully achieve what you want, it looks like you only want to check the versions of those listed dependencies, and no others?

```json
{
  "versionGroups": [
    {
      "dependencies": ["@types/react", "@typescript-eslint*", "react-dom", "react", "typescript"],
      "packages": ["**"]
    },
    {
      "dependencies": ["**"],
      "packages": ["**"],
      "isIgnored": true
    }
  ]
}
```

Notice the `*` after `"@typescript-eslint*"` to match anything that starts with `@typescript-eslint`

Comment (jonathansadowski) [reply .syncpackrc:3]: Thanks @JamieMason, that's exactly what we want, and a much more manageable way of maintaining it. I'll give it a shot in a few!

Comment (JamieMason) [reply .syncpackrc:3]: I forgot to mention that with this in place you can delete the `filter` property as it won't be needed. I put [a note in the readme about filter](https://github.com/JamieMason/syncpack/#filter) but that was only fairly recently, I would avoid using it if you can.

**Pull request merged**